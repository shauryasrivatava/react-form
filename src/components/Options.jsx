import React from 'react';
import PropTypes from 'prop-types';

const Options = (props) => {
    var element = props.options.map((item, index) => {
        return <option key={index} value={item}>{item}</option>
    })
    return element
}

Options.propTypes = {
    options: PropTypes.array
};

Options.defaultProps = {
    options: []
};

export default Options;