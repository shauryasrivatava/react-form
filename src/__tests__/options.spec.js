import renderer from 'react-test-renderer';
import Options from '../components/Options';
import React from 'react';
import { mount } from 'enzyme';
import enzymeConfig from '../../enzymeConfig'

const options = ['English', 'Spanish', 'German', 'French'];
describe('<Options />', function(){

    it('should render correctly', function(){
        const component = renderer.create(<Options options={options}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have the correct number of option tags', function(){
        const component = mount(<Options options={options} />);
        expect(component.find('option').length).toEqual(4);
    });

    it('should render the correct option names', function() {
        let languages = '';
        const expected = 'EnglishSpanishGermanFrench';
        const component = mount(<Options options={options} />);
        let optionArray = component.find('option');

        optionArray.forEach((item) => {
            languages += item.text();
        })
        expect(languages).toEqual(expected);
    });
})
