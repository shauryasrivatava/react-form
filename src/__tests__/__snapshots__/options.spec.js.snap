// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> should render correctly 1`] = `
Array [
  <option
    value="English"
  >
    English
  </option>,
  <option
    value="Spanish"
  >
    Spanish
  </option>,
  <option
    value="German"
  >
    German
  </option>,
  <option
    value="French"
  >
    French
  </option>,
]
`;
